static/js/message_edit.js

```diff
@@ -585,6 +595,37 @@ exports.save_inline_topic_edit = function (row) {
     });
 };
 
+function maybe_move_to_new_thread(opts) {
```

Comment:
This should be a separate commit; I'm not sure it's a good idea done this way in any case.  We have logic in `message_events.js` for handling the edit to renarrow if appropriate already, which this duplicates.  The change that might make sense to make is to adjust the existing code in `message_events.js` to renarrow even when it was just an "All following" if it indeed moved all messages in the narrow.

I dropped this from the PR for now.